File: DevExpress VCL/ExpressQuantumGrid 7/Demos/Delphi/ViewCardDemo/ViewCardDemoData.pas
unit ViewCardDemoData;

interface

uses
  Forms, SysUtils, Classes, DB, cxStyles, ImgList,
  Controls, cxEdit, cxEditRepositoryItems, cxDBEditRepository, cxClasses,
  cxGridCardView, DBTables;

type
  TViewCardDemoDataDM = class(TDataModule)
    DataBase: TDataBase;
    tblPersons: TTable;
    dsPersons: TDataSource;
    tblPersonsID: TAutoIncField;
    tblPersonsFIRSTNAME: TStringField;
    tblPersonsSECONDNAME: TStringField;
    tblPersonsGENDER: TBooleanField;
    tblPersonsBIRTHNAME: TStringField;
    tblPersonsDATEOFBIRTH: TDateTimeField;
    tblPersonsBIRTHCOUNTRY: TIntegerField;
    tblPersonsLOCATIONOFBIRTH: TStringField;
    tblPersonsBIOGRAPHY: TMemoField;
    tblPersonsNICKNAME: TStringField;
    tblPersonsFullName: TStringField;
    ilPics: TImageList;
    tblPersonsHOMEPAGE: TStringField;
    tblCountries: TTable;
    dsCountries: TDataSource;
    procedure tblPersonsCalcFields(DataSet: TDataSet);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  ViewCardDemoDataDM: TViewCardDemoDataDM;

implementation

{$R *.dfm}

procedure TViewCardDemoDataDM.tblPersonsCalcFields(
  DataSet: TDataSet);
begin
  with DataSet do
    tblPersonsFullName.AsString := Format('%s %s (%d)',[tblPersonsFirstName.AsString,
      tblPersonsSecondName.AsString, tblPersonsID.AsInteger]);
end;

end.

File: DevExpress VCL/ExpressQuantumGrid 6/Demos/CBuilder/ViewCardDemo/ViewCardDemoData.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "ViewCardDemoData.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxClasses"
#pragma link "cxGridTableView"
#pragma link "cxStyles"
#pragma link "cxDBEditRepository"
#pragma link "cxEdit"
#pragma link "cxEditRepositoryItems"
#pragma link "cxGridCardView"
#pragma resource "*.dfm"
TViewCardDemoDataDM *ViewCardDemoDataDM;
//---------------------------------------------------------------------------
__fastcall TViewCardDemoDataDM::TViewCardDemoDataDM(TComponent* Owner)
  : TDataModule(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TViewCardDemoDataDM::tblPersonsCalcFields(
      TDataSet *DataSet)
{
  tblPersonsFullName->AsString = tblPersonsFIRSTNAME->AsString + " " +
    tblPersonsSECONDNAME->AsString + " " + "(" +
    IntToStr(tblPersonsID->AsInteger) + ")";
}
//---------------------------------------------------------------------------

